data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/EventHandle.java

```diff
@@ -14,4 +17,46 @@ public interface EventHandle {
      * @since 2.2
      */
     void release(boolean result);
+
+    /**
+     * sets acknowledgement set
+     *
+     * @param acknowledgementSet acknowledgementSet to be set in the event handle
+     * @since 2.6
+     */
+    void setAcknowledgementSet(final AcknowledgementSet acknowledgementSet);
```

Comment:
Do we need to expose all of this? It might make sense to have a different interface for internal use. Say `InternalEventHandle`. Then `DefaultEventHandle` can implement that.

Reply:
Not sure I understand this suggestion. Are you suggesting we have `InternalEventHandle` and `EventHandle` separately?

Reply:
Yes, we'd have two interfaces:

```
public interface InterfaceEventHandle {
...
// methods that data-prepper-core calls are here.
}
```

And we implement it as:

```
class DefaultEventHandle implements EventHandle, InternalEventHandle
{
...
}
```

In data-prepper, core we'd need a cast: `InternalEventHandle internalEventHandle = (InternalEventHandle) eventHandle`.

This would be a good indication that somebody is doing something they shouldn't if we see it outside of data-prepper-core.